internal/testutils/services.go

```diff
@@ -0,0 +1,17 @@
+package testutils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+)
+
+func PrepareRequest(ctx context.Context, method string, path string, data *map[string]interface{}) (req *http.Request, err error) {
+	json_data, err := json.Marshal(data)
+	if err != nil {
+		return
+	}
+	req, err = http.NewRequestWithContext(ctx, "POST", "/api/provisioning/pubkeys", bytes.NewBuffer(json_data))
+	return
+}
```

Comment:
This literally saves a single line plus one error handling, I think for tests it's safe to ignore the error (since the input is constant). You could avoid this function completely including the testutils package alltogether which I am trying to hate :-)

I value readability of tests to short code, also over time we will see functions like PrepareRequestWithXYZ etc and this can get out of control.

Reply:
I'm totally agree, it was me overengineering the hell out of it, lets not :+1: